src/credentialstore/credentialstore.ts

```diff
@@ -26,44 +44,141 @@ export class CredentialStore implements ICredentialStore {
+            }
+            return success;
+        } catch (err) {
+            throw(err);
+        }
+    }
+
+    // Native Credential implementations
+
+    private async nativeReadCredential(credentialId: string, cred: Contracts.Credential): Promise<Contracts.Credential> {
+        try {
+            const savedPassword: string = await this._context.secrets.get(credentialId);
+            cred.password = savedPassword;
+            return cred;
+        } catch (err) {
+            throw(err);
+        }
+    }
+
+    private async nativeSaveCredential(credentialId: string, password: any): Promise<boolean> {
+        try {
+            await this._context.secrets.store(credentialId, password);
+            const savedPassword = await this._context.secrets.get(credentialId);
+            if (savedPassword === password) {
+                return true;
+            }
+            return false;
+        } catch (err) {
+            throw(err);
+        }
+    }
+
+    private async nativeDeleteCredential(credentialId: string): Promise<boolean> {
+        try {
+            await this._context.secrets.delete(credentialId);
+            const savedPassword = await this._context.secrets.get(credentialId);
+            if (!savedPassword) {
+                return true;
+            }
+            return false;
+        } catch (err) {
+            throw(err);
+        }
+    }
+
+    /**
+     *
+     * @returns Migrates all saved credentials to the native credential system
+     */
+    private async migratePasswords(): Promise<void> {
+        const connections = vscode.workspace.getConfiguration(Constants.extensionName).get<any[]>(Constants.connectionsArrayName);
+        const savedPasswordConnections = connections.filter(conn => conn.savePassword === true);
+        for (let i = 0; i < savedPasswordConnections.length; i++) {
+            let conn = savedPasswordConnections[i];
+            await this.cleanCredential(conn);
+        }
+        await Utils.removeCredentialFile();
+    }
+
+
+    private async cleanCredential(conn): Promise<boolean> {
+        const credentialId = ConnectionStore.formatCredentialId(conn);
+        const credential = await this._client.sendRequest(Contracts.ReadCredentialRequest.type, { credentialId, password: undefined });
+        if (credential.password) {
+            const password = credential.password;
+            // save it in secret store
+            await this._secretStorage.store(credentialId, password);
+            // check if it's saved
+            const savedPassword = await this._secretStorage.get(credentialId);
+            if (savedPassword === password) {
+                // delete from tools service
+                const result = await this._client.sendRequest(Contracts.DeleteCredentialRequest.type,
+                    { credentialId, password: undefined });
+                return result;
+            } else {
+                return false;
```

Comment:
Anywhere you're returning false should be an error thrown - if we can't successfully migrate all the passwords it will currently just silently skip them which doesn't seem ideal. Throwing an error and letting the user decide if they care seems like a better way to handle that.

Reply:
The user wouldn't know the passwords are migrated, since it's always done. What suggestion do you have to ask users in case there was an error here?